lib/charliecloud.py

```diff
@@ -210,12 +220,36 @@ def __init__(self, ref, download_cache, unpack_dir, image_subdir=None):
    def __str__(self):
       return str(self.ref)
 
+   @property
+   def default_arch_dlcache(self):
+      return os.path.dirname(os.path.dirname(
+                             self.download_cache)) + '/amd64/dlcache'
+
+   @property
+   def default_arch_unpack(self):
+      return os.path.dirname(os.path.dirname(
+                             self.download_cache)) + '/amd64/img'
+
+   @property
+   def inferred_fat_dir(self):
+      return os.path.dirname(os.path.dirname(self.download_cache)) + '/fatman'
```

Comment:
Why do we need separate directories in the download cache for different architectures?

Reply:
To avoid clashes with manifest files that have the same name but different content (different architectures). Alternatively we can just change our manifest storage convention to specify architecture, e.g., `centos:8.arm64%v8.manifest.json`.